src/main/java/org/concordion/internal/command/strategies/BestMatch.java

```diff
@@ -0,0 +1,84 @@
+package org.concordion.internal.command.strategies;
+
+import org.concordion.api.CommandCall;
+import org.concordion.api.ElementUpdating;
+import org.concordion.api.Evaluator;
+import org.concordion.api.ResultRecorder;
+import org.concordion.api.listener.VerifyRowsListener;
+import org.concordion.internal.Row;
+import org.concordion.internal.SummarizingResultRecorder;
+import org.concordion.internal.command.VerifyRowsCommand;
+import org.concordion.internal.util.Announcer;
+
+public class BestMatch extends VerifyRowsCommand.VerificationStrategy {
+
+    public BestMatch(CommandCall commandCall, Evaluator evaluator, ResultRecorder resultRecorder,
+                     Announcer<VerifyRowsListener> listeners, String loopVariableName, Iterable<Object> actualRows) {
+        super(commandCall, evaluator, resultRecorder, listeners, loopVariableName, actualRows);
+    }
+
+    @Override
+    public void verify() {
+        announceExpressionEvaluated(commandCall.getElement());
+        for (Row expectedRow : expectedRows) {
+            tableSupport.copyCommandCallsTo(expectedRow);
+            Object row = findBestMatchingRow();
+            if (row != null) {
+                evaluator.setVariable(loopVariableName, row);
+                commandCall.getChildren().verify(evaluator, resultRecorder);
+                actualRows.remove(row);
+            } else {
+                announceMissingRow(expectedRow.getElement());
+            }
+        }
+        reportSurpulusRows();
```

Comment:
It's a small thing, but you've got one extra "u" in surplus. I noticed this in some of the specs too. Would you fix the spelling please.